src/commands/distribute/release.ts

```diff
@@ -441,25 +452,32 @@ export default class ReleaseBinaryCommand extends AppCommand {
   }
 
   private async loadReleaseIdUntilSuccess(app: DefaultApp, uploadId: string): Promise<any> {
+    const t0 = Date.now();
+    const t1 = t0 + (Number.parseInt(this.timeout, 10) >= 0 ? Number.parseInt(this.timeout, 10) * 1000 : 0);
     return new Promise((resolve, reject) => {
-      const timerId = setInterval(async () => {
+      const check = async () => {
         let response;
         try {
           response = await this.loadReleaseId(app, uploadId);
         } catch (error) {
-          clearInterval(timerId);
           reject(new Error(`Loading release id failed with error: ${error.errorMessage}`));
         }
         const releaseId = response.release_distinct_id;
         debug(`Received release id is ${releaseId}`);
         if (response.upload_status === "readyToBePublished" && releaseId) {
-          clearInterval(timerId);
+          debug(`Loading release id completed, total time: ${(Date.now() - t0) / 1000}`);
           resolve(Number(releaseId));
         } else if (response.upload_status === "error") {
-          clearInterval(timerId);
+          debug(`Loading release id completed, total time: ${(Date.now() - t0) / 1000}`);
           reject(new Error(`Loading release id failed: ${response.error_details}`));
+        } else if (t1 > t0 && Date.now() >= t1) {
+          debug(`Loading release id completed, total time: ${(Date.now() - t0) / 1000}`);
```

Comment:
Perhaps we don't need this `debug` as the error below tells us the required information.